apps/communications/contacts/js/views/selector.js

```diff
@@ -0,0 +1,112 @@
+  selectorHeader.addEventListener('action', cancelHandler);
+  container.addEventListener('click', clickHandler);
+
+  var contactsToImport = [];
+
+  function cancelHandler() {
+    selectorHeader.removeEventListener('action', cancelHandler);
+    saveButton.removeEventListener('click', importAll);
+    container.removeEventListener('click', clickHandler);
+    window.postMessage({
+      type: 'abort',
+      data: ''
+    }, fb.CONTACTS_APP_ORIGIN);
+    Contacts.cancel();
+  }
+
+  function clickHandler(event) {
+    // TODO Launch activity {mozContactParam: true},  passing contact to form.
+    console.log(event.target);
+  }
+
+  function importAll() {
+    var numDupsMerged = 0,
+        importedContacts = 0;
+
+    contactsToImport.forEach((contact, index) => {
+      contacts.Matcher.match(contact, 'passive', {
+        onmatch: matches => {
+          var callbacks = {
+            success: function(mergedContact) {
+              numDupsMerged++;
+              _doContinue(index, true);
+            },
+            error: _doContinue.bind(this, index, true)
+          };
+          contacts.adaptAndMerge(contact, matches, callbacks);
+        },
+        onmismatch: function() {
+          var saving = navigator.mozContacts.save(contact);
+
+          saving.onsuccess = () => {
+            _doContinue(index);
+          };
+
+          saving.onerror = (err) => {
+            console.error(err);
+            _doContinue(index, true);
+          };
+        }
+      });
+    });
+
+    function _doContinue(index, failed) {
+      if (!failed) {
+        importedContacts++;
+      }
+
+      if (contactsToImport.length - 1 <= index) {
+        utils.status.show({
+          id: 'memoryCardContacts-imported3', // FIXME show better message.
+          args: {n: importedContacts}
+        }, {
+          id: 'contactsMerged',
+          args: {numDups: numDupsMerged}
+        });
+
+        window.setTimeout(cancelHandler, 1500); // Allows to see status message.
+      }
+    }
+  }
+
+  var addContact = contacts => {
+    if (!Array.isArray(contacts)) {
+      contacts = [contacts];
+    }
+    contacts.forEach( _doAddContact);
+  };
+
+  var _doAddContact = function(contact) {
+    var template = document.getElementById('contactToImportTemplate');
+    var clone = document.importNode(template.content, true);
+
+    var contactData = clone.querySelectorAll('p');
+    contactData[0].textContent = contact.name[0];
+    contactData[1].textContent = contact.tel[0].value;
+
+    if (contact.photo) {
+      var picture = clone.querySelector('aside span');
+      var reader = new FileReader();
+      reader.onloadend = () =>  {
+        picture.setAttribute('style', 'background: url(' + reader.result + ')');
+      };
+      reader.readAsDataURL(contact.photo[0]);
+    }
+
+    container.appendChild(clone);
+    contactsToImport.push(contact);
+  };
+
+  return {
+    addContact: addContact
+  };
+})(document, window, contacts, fb, utils);
```

Comment:
nit: in order to be consistent with the rest of modules I would suggest we do not pass these objects as parameters to the closure